.github/workflows/ci.yml

```diff
@@ -116,6 +116,35 @@ jobs:
         ./scripts/run_tests.sh
         deactivate
       shell: bash
+  
+  build_with_all_flags:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.12']
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    
+    - name: Configure and build project with all settings are ON
+      run: |
+        sudo apt update -y
+        sudo bash -x dependencies.sh -y
+        mkdir build
+        cd build
+        cmake .. -DUSE_ETCD=ON -DUSE_ETCD_LEGACY=ON -DUSE_REDIS=ON -DUSE_HTTP=ON -DWITH_STORE=ON -DWITH_P2P_STORE=ON
```

Comment:
The `-DUSE_ETCD_LEGACY=ON` option cannot be enabled, as the build for the legacy version of ETCD has already been removed from the CI.